File: app/src/main/java/info/nightscout/androidaps/plugins/general/nsclient/acks/NSAuthAck.java
package info.nightscout.androidaps.plugins.general.nsclient.acks;

import org.json.JSONObject;

import info.nightscout.androidaps.events.Event;
import info.nightscout.androidaps.plugins.bus.RxBus;
import io.socket.client.Ack;

public class NSAuthAck extends Event implements Ack{
    public boolean read = false;
    public boolean write = false;
    public boolean write_treatment = false;

    public void call(Object...args) {
        JSONObject response = (JSONObject)args[0];
        read = response.optBoolean("read");
        write = response.optBoolean("write");
        write_treatment = response.optBoolean("write_treatment");
        RxBus.INSTANCE.send(this);
    }
}

File: app/src/main/java/info/nightscout/androidaps/plugins/sync/nsclient/acks/NSAuthAck.kt
package info.nightscout.androidaps.plugins.sync.nsclient.acks

import info.nightscout.androidaps.events.Event
import info.nightscout.androidaps.plugins.bus.RxBus
import io.socket.client.Ack
import org.json.JSONObject

class NSAuthAck(private val rxBus: RxBus) : Event(), Ack {

    var read = false
    var write = false
    var writeTreatment = false
    override fun call(vararg args: Any) {
        val response = args[0] as JSONObject
        read = response.optBoolean("read")
        write = response.optBoolean("write")
        writeTreatment = response.optBoolean("write_treatment")
        rxBus.send(this)
    }
}